From 45a7c623bce15bb558eb1121b528eff2f99fa6a8 Mon Sep 17 00:00:00 2001
From: Guy Nisan <gnisan@habana.ai>
Date: Sun, 15 Nov 2020 14:02:28 +0200
Subject: [PATCH] [SW-14676] fwupd: clear status fast

Clear status command is not slow as it takes a short time to execute and
it does not need to wait-for-clear. Check out this command as fast will
prevent from other modules to go through disable-run-enable routine.

Change-Id: Id369bc01c66f82a7f7ebe3f95aac812760889260
Signed-off-by: Guy Nisan <gnisan@habana.ai>
---

diff --git a/preboot/module_fwupd.c b/preboot/module_fwupd.c
index 9d52bed..523fd94 100644
--- a/preboot/module_fwupd.c
+++ b/preboot/module_fwupd.c
@@ -40,6 +40,7 @@
 	cmd.val = regmap_readl(mmUPD_CMD);
 	switch (cmd.cmd) {
 	case HL_UPD_CLR_STS:
+		return FOUND_FAST_RETURN;
 	case HL_UPD_RST_STATE:
 	case HL_UPD_MAP_SRAM:
 	case HL_UPD_RUN_SRAM:
diff --git a/preboot/preboot.c b/preboot/preboot.c
index a4e2892..fd4412b 100644
--- a/preboot/preboot.c
+++ b/preboot/preboot.c
@@ -38,7 +38,7 @@
 		.disable = noop,
 		.run = module_fwupd_run,
 		.enable = noop,
-		.run_fast = noop,
+		.run_fast = module_fwupd_run,
 		.teardown = noop,
 		.run_noreturn = module_fwupd_run_noret,
 	},
